scipy/special/specfun/specfun.f

```diff
@@ -8738,8 +8738,9 @@ SUBROUTINE E1Z(Z,CE1)
 10         CONTINUE
 15         CONTINUE
            IF (X.LE.0.0.AND.DIMAG(Z).EQ.0.0) THEN
-C             Careful on the branch cut -- avoid signed zeros
-              CE1=-EL-CDLOG(-Z)+Z*CE1-PI*(0.0D0,1.0D0)
+C     Careful on the branch cut -- use the sign of the imaginary part
+C     to get the right sign on the factor if pi.
+              CE1=-EL-CDLOG(-Z)+Z*CE1-DSIGN(PI,DIMAG(Z))*(0.0D0,1.0D0)
```

Comment:
This may also affect the behavior of the complex expi, best to check that code also